src/phy/mod.rs

```diff
@@ -132,6 +137,33 @@ pub use self::tracer::Tracer;
 ))]
 pub use self::tuntap_interface::TunTapInterface;
 
+/// An ID that can be used to uniquely identify a packet to a [`Device`],
+/// sent or received by that same [`Device`]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[derive(Debug, PartialEq)]
+pub struct PacketId {
+    id: usize,
+}
+
+impl PacketId {
+    pub const fn none() -> Option<Self> {
+        None
+    }
```

Comment:
Not sure if this belongs here. If we really need a `None` then we can just write `None` right? Is there a case where you need this maybe?

Reply:
Similar to above, because `PacketMetadata` is no longer `Copy`, the `const fn` makes it "easier" to just call `PacketMetadata::none()`. Not so user-friendly, and `PacketMetadata` will probably become `Copy` again, in which case this can be removed